party_time/party_time.lua

```diff
@@ -0,0 +1,345 @@
+                ui.location(93,72)
+                if ui.button('decline', 'Decline') then
+                    command.input("/decline")
+                    unhandeled_invite = false
+                end
+                    
+            end)
+            if invite_dialog.closed then
+                invite_dialog.closed = nil
+                unhandeled_invite = false
+            end
+            if invite_dialog.state.x ~= pt_settings.ui.x or invite_dialog.state.y ~= pt_settings.ui.y then
+                pt_settings.ui.x = invite_dialog.state.x
+                pt_settings.ui.y = invite_dialog.state.y
+                invite_dialog_state.x = invite_dialog.state.x
+                invite_dialog_state.y = invite_dialog.state.y
+                settings.save(pt_settings)
+            end
+        end
+
+        local closed_dialogs = {}
+        for id, request_dialog in pairs(unhandled_requests) do 
+            request_dialog.state, request_dialog.close = ui.window(id, request_dialog.state, function()
+                ui.location(11, 5)
+                ui.text(id .. ' has requested\nto join your party')
+                
+                ui.location(11, 50)
+                if pt_settings.auto.accept then
+                    if ui.check('add_to_whitelist', 'Remember ' .. id, request_dialog.add_to_whitelist) then
+                        request_dialog.add_to_whitelist = not request_dialog.add_to_whitelist
+                    end
+                else
+                    if ui.check('add_to_whitelist', 'Turn auto accept on ', pt_settings.auto.accept) then
+                        pt_settings.auto.accept = true
+                    end
+                end
+
+                ui.location(11,72)
+                if ui.button('invite', 'Invite') then
+                    command.input("/pcmd add " .. id)
+                    closed_dialogs[#closed_dialogs + 1] = id
+                    if request_dialog.add_to_whitelist then
+                        print(id)
+                        pt_settings.whitelist[id] = true
+                        settings.save(pt_settings)
+                    end
+                end
+                ui.location(93,72)
+                if ui.button('ignore', 'Ignore') then
+                    closed_dialogs[#closed_dialogs + 1] = id
+                end
+            end)
+            if request_dialog.close then
+                closed_dialogs[#closed_dialogs + 1] = id
+            end
+            if request_dialog.state.x ~= pt_settings.ui.x or request_dialog.state.y ~= pt_settings.ui.y then
+                pt_settings.ui.x = request_dialog.state.x
+                pt_settings.ui.y = request_dialog.state.y
+                invite_dialog_state.x = request_dialog.state.x
+                invite_dialog_state.y = request_dialog.state.y
+                settings.save(pt_settings)
+            end
+        end
+
+        for _, id in pairs(closed_dialogs) do
+            unhandled_requests[id] = nil
+        end
```

Comment:
Sets would work great here too! `/super subtle hint`

```lua
        unhandled_requests:difference(closed_dialogs)
```

Sets are nifty.

Reply:
I am not sure i see how this one would work out.
an unhandled_request looks like ```lua
        unhandled_requests[p.player_name] = {
            state = {
                title = 'Party Request',
                style = 'normal',
                x = pt_settings.ui.x,
                y = pt_settings.ui.y,
                width = 179,
                height = 96,
                resizable = false,
                moveable = true,
                closable = true,
            },
            add_to_whitelist = false,
        }
```
and closed_dialogs has the index of the closed request.
how would the set know that the dialog in closed_dialogs is the same as the one in unhandled_requests to perform the difference?